### DISCUSSION TITLE
One language (default)

### DISCUSSION THREAD

**Discussion by icifra** [McShelby/hugo-theme-relearn#475] [Q&A] [answered]
Hello, this theme is definitely one of the best in the collection of themes, it looks great on both computer and mobile. Extensive and detailed documentation (although it can be quite a steep curve for a beginner).

I would like to try this theme, give me some tips at the start:
- is it better to install it as a Hugo module? (will future updates break my site?);
- how to immediately set up a theme only for my language other than English (without using multilingualism, i.e. my default language), and features of building a site in this case.

Thanks in advance for your help!

**Comment by cagix** [comment]
> is it better to install it as a Hugo module?

I was wondering the same thing. What is the recommended installation method?

**Comment by McShelby** [comment, marked as answer]
The **installation method** depends on your preferences and requirements. There is no common recommendation.

Eg. in my company we do not have a Go infrastructure (not even git), so we are copying the theme files over into our own project.

The theme tries to adhere to semver, so you can have certain expectations on stability.

Since a few days, there are also tags available in the git repository to pin your module to a ceratain version. Tho
se tags may also be usable using Go modules but I am no expert on this. See https://mcshelby.github.io/hugo-theme-relearn/basics/installation/index.html#from-git

To **configure without multilang** settings you can take the `config.toml` of the exampleSite and remove all language specific settings or just remove all but one language. There are many ways to configure this, all documented in the [official Hugo docs](http://gohugo.io)

**Documentation** is still an issue. I am also not happy with the current structure but it is really time consuming to restructure them.

**Comment by icifra** [comment]
I tried just removing all languages from `i18n`.  Although this is rough), but it should work.

The easiest way (in my initial opinion) is to tweak the following settings:
1. Set your default language (be sure to set languages 1 (custom) and 2 (English)
`defaultContentLanguage = "de"`
2. Disable language switching display
 `disableLanguageSwitchingButton=true`
4. Customize some shortcodes for your language (let's say a link to Github)
`[[Languages.de.menu.shortcuts]]`

This option is good because it does not break anything guaranteed

**Comment by McShelby** [comment]
This sounds like you are tweaking the `exampleSite` settings. Although many theme users are doing this, I strongly **don't recommend** this.

Yes, it lets you start quickly, but it makes it really difficult once you want to update the theme to the latest version, as the exampleSite and its configuration will usually change between releases and will generate merge conflicts on your side.

In other words: **If you are editing _any_ of the released files, you are doing something wrong!**

Or you are fixing a bug. In this case, a pull request or issue report is recommended.

Instead I recommend one of the usual ways to install a theme into your Hugo project:

- manually copy a [theme release](https://github.com/McShelby/hugo-theme-relearn/releases) over to your project's `themes` directory
- [use git to install a theme](https://gohugobrasil.netlify.app/themes/installing-and-using-themes/)
- [use Hugo modules to install a theme](https://gohugo.io/hugo-modules/use-modules/#use-a-module-for-a-theme)

Remember, using git or Hugo modules without a specific tag will install the latest changes from the GitHub main branch, which might be a) unstable b) change pretty often. It's better to use a specific version (wildcards in the form of `x`, `1.x` and `4.5.x`  are supported.

**Comment by icifra** [comment]
No no, I'm configuring the file in my file `config.toml`. 

And this is a setting without removing the rest of the languages. I think this is the most correct way for one default language
